src/DumpCallMap.hs

```diff
@@ -1,157 +1,22 @@
 {-# language ScopedTypeVariables #-}
 module DumpCallMap (encodeCallMap, hPutCallMap) where
 
-import Types (CallMap(..)
-             , CallTree(..)
-             , PermissionAction(..)
-             , PermissionActionSet
-             , PermissionName(..)
-             , permissionActionName
-             )
+import Types (CallMap(..))
 import Control.Monad.IO.Class (MonadIO(..))
 
-import qualified Data.ByteString.Builder as B
-import Data.Foldable
-import qualified Data.Map as Map
-import Data.List (intersperse)
-import Data.Monoid (Monoid(..), (<>))
-import qualified Data.Text.Encoding as DTE
+import qualified Data.ByteString.Lazy as B
 import System.IO (Handle, hFlush)
+import qualified Data.Aeson as Aeson
 
-import Language.C.Data.Ident (Ident, identToString)
-import Language.C.Data.Node (NodeInfo, CNode(nodeInfo))
-import Language.C.Data.Name (Name)
-import qualified Language.C.Data.Node as Node
-import qualified Language.C.Data.Position as Pos
-
-type CallMapItem = (NodeInfo, CallTree Ident, PermissionActionSet)
-
--- | Encode the given callmap as a "Data.ByteString
-encodeCallMap :: CallMap -> B.Builder
-encodeCallMap (CallMap cm) = form "callmap" True $ Map.foldMapWithKey encodeCallMapItem cm
```

Comment:
*In principle* from a release management perspective I like to retain support for old formats, but honestly I don’t think anyone is really depending on this, and a user can just blow away and regenerate the files. So ±0 on that.

Reply:
While I can understand that I think the benefits of this format are *far* outweighed by the persistent maintenance cost that it incurs (not to mention the fact that it showed up pretty heavily in my profiling, so there was clearly some performance on the table).

Reply:
Agreed, just getting this down for posterity.